### PR TITLE
[Core] Add `NO_UPLOAD` for `remote_identity`

### PR DESCRIPTION
**Pull request by romilbhardwaj**
User reported they are using an external mechanism to inject AWS credentials in their k8s pods, which should be used for accessing buckets. However, we upload AWS credentials from the local machine if they exist, and as a result these local user credentials were overriding the injected credentials.

This PR adds a `NO_UPLOAD` flag to `remote_identity`, to explicitly skip uploading cloud credentials even if they exist on the local machine.

- [x] Tested on a SkyPilot cluster launched on k8s, with `aws.remote_identity: NO_UPLOAD` and verified AWS credentials are not uploaded.

### CODE REVIEW

File: sky/utils/schemas.py [hunk 1]
```diff
@@ -655,6 +655,7 @@ class RemoteIdentityOptions(enum.Enum):
     """
     LOCAL_CREDENTIALS = 'LOCAL_CREDENTIALS'
     SERVICE_ACCOUNT = 'SERVICE_ACCOUNT'
+    SKIP = 'SKIP'
```
Comment (romilbhardwaj) [inline sky/utils/schemas.py:658]: Open to better names, if any

Comment (Michaelvll) [reply sky/utils/schemas.py:658]: How about `remote_identity: null`?

Comment (concretevitamin) [reply sky/utils/schemas.py:658]: Being explicit makes it more clear to me, how about 'NO_UPLOAD'?

Comment (romilbhardwaj) [reply sky/utils/schemas.py]: Thanks, renamed to `NO_UPLOAD`

File: sky/backends/backend_utils.py [hunk 2]
```diff
@@ -706,6 +706,15 @@ def write_cluster_config(
...
         excluded_clouds = [cloud]
+
+    for cloud_str, cloud_obj in cloud_registry.CLOUD_REGISTRY.items():
+        remote_identity_config = skypilot_config.get_nested(
+            (cloud_str.lower(), 'remote_identity'), None)
+        if remote_identity_config:
+            if (remote_identity_config ==
+                    schemas.RemoteIdentityOptions.NO_UPLOAD.value):
+                excluded_clouds.append(cloud_obj)
+
```
Comment (Michaelvll) [inline sky/backends/backend_utils.py:717]: The error message raised above can be misleading if a user specifies `NO_UPLOAD` for the current cloud. Should we update that?

Comment (romilbhardwaj) [reply sky/backends/backend_utils.py]: Ah thanks, updated now

File: sky/backends/backend_utils.py [hunk 3]
```diff
@@ -699,13 +699,29 @@ def write_cluster_config(
...
-                '`remote_identity: LOCAL_CREDENTIALS`.')
-        excluded_clouds = [cloud]
+        if remote_identity == schemas.RemoteIdentityOptions.NO_UPLOAD.value:
+            # If NO_UPLOAD is specified, fall back to default remote identity
+            # for downstream logic but add it to excluded_clouds to skip
+            # credential file uploads.
+            remote_identity = schemas.get_default_remote_identity(
+                str(cloud).lower())
+        else:
+            if not cloud.supports_service_account_on_remote():
```
Comment (Michaelvll) [inline sky/backends/backend_utils.py:709]: nit:
```suggestion
        elif not cloud.supports_service_account_on_remote():
```

File: docs/source/reference/config.rst [hunk 4]
```diff
@@ -244,6 +244,10 @@ Available fields and semantics:
     # instances. SkyPilot will auto-create and reuse a service account (IAM
     # role) for AWS instances.
     #
+    # NO_UPLOAD: No credentials will be uploaded to the pods. Useful for
+    # avoiding overriding any existing credentials that may be automounted on
+    # the cluster.
```
Comment (Michaelvll) [inline docs/source/reference/config.rst:249]: Can we update the caveats below for `SERVICE_ACCOUNT` L268, i.e. if people need fully disable the upload, they should use `NO_UPLOAD`.

File: sky/backends/backend_utils.py [hunk 5]
```diff
@@ -699,13 +699,29 @@ def write_cluster_config(
...
+        excluded_clouds.add(cloud)
+
+    for cloud_str, cloud_obj in cloud_registry.CLOUD_REGISTRY.items():
+        remote_identity_config = skypilot_config.get_nested(
+            (cloud_str.lower(), 'remote_identity'), None)
+        if remote_identity_config:
+            if (remote_identity_config ==
+                    schemas.RemoteIdentityOptions.NO_UPLOAD.value):
+                excluded_clouds.add(cloud_obj)
+
```
Comment (Michaelvll) [inline sky/backends/backend_utils.py:724]: I found the logic for the remote identities is getting a bit complicated now. Can we add a comment at the top talking about the four cases: `LOCAL_CREDENTIALS`, `SERVICE_ACCOUNT`, `custom service account name`  and `NO_UPLOAD`

**Review by Michaelvll — APPROVED** [review]
Thanks @romilbhardwaj! Left some minor comments.

File: sky/backends/backend_utils.py (hunk 5 above)
Comment (romilbhardwaj) [reply sky/backends/backend_utils.py:724]: Good point - updated comments

**Pull request merged**